worker/docker_image_manager.py

```diff
@@ -0,0 +1,135 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    Warning: Assumes that Docker images are stored on the same disk as the
+    work directory specified by the `workdir` parameter. If this is not the
```

Comment:
Oops that isn't necessary, I can just point `statvfs()` to `/var/lib/docker`, and it looks like you don't need root permissions to do that (just tried it on one of the Azure vms).